manifests/filter/header_user.pp

```diff
@@ -0,0 +1,77 @@
+# == Resource: repose::filter::header_user
+#
+# This is a resource for generating header-user configuration files
+#
+# === Parameters
+#
+# [*ensure*]
+# Bool. Ensure config file present/absent
+# Defaults to <tt>present</tt>
+#
+# [*filename*]
+# String. Config filename
+# Defaults to <tt>header-user.cfg.xml</tt>
+#
+# [*source-headers*]
+# Array of hashes containing source headers with the ids of the source headers and their
+# coresponding values
+# Example:
+# [
+#   {
+#     'X-Auth-Token' => '.90',
+#     'X-PP-User'    => '.07',
+#     'X-User-Name'  => '.03',
+#   },
+# ]
+#
+# === Links
+#
+# *
+#
+# === Authors
+#
+# * Senthil Natarajan <mailto:senthil.natarajan@rackspace.com>
+# * Joel Rizner <mailto:joel.rizner@rackspace.com>
+# * c/o Cloud Integration Ops <mailto:cit-ops@rackspace.com>
+#
+define repose::filter::header_user (
+  $ensure         = present,
+  $filename       = 'header-user.cfg.xml',
+  $source_headers = undef
```

Comment:
Since this is a required parameter, we should remove the default, so change it to:

```
  $source_headers,
```